### PR TITLE
JCLOUDS-993 - Add missing parameters to `VirtualMachineApi`

### PR DESCRIPTION
**Pull request by momania**
- `serviceOfferingId` at `#changeServiceForVirtualMachine`
- `options` of type `UpdateVirtualMachineOptions` at
  `#updateVirtualMachine`

### CODE REVIEW

File: apis/cloudstack/src/test/java/org/jclouds/cloudstack/features/VirtualMachineApiTest.java [hunk 1]
```diff
@@ -204,11 +205,14 @@ public void testChangeServiceForVirtualMachine() throws SecurityException, NoSuc
    }
 
    public void testUpdateVirtualMachine() throws SecurityException, NoSuchMethodException, IOException {
-      Invokable<?, ?> method = method(VirtualMachineApi.class, "updateVirtualMachine", String.class);
-      GeneratedHttpRequest httpRequest = processor.createRequest(method, ImmutableList.<Object> of(5));
+      Invokable<?, ?> method = method(VirtualMachineApi.class, "updateVirtualMachine", String.class,
+              UpdateVirtualMachineOptions.class);
+      GeneratedHttpRequest httpRequest = processor.createRequest(method, ImmutableList.<Object> of("5",
+              UpdateVirtualMachineOptions.Builder.group("test")));
```
Comment (nacx) [inline apis/cloudstack/src/test/java/org/jclouds/cloudstack/features/VirtualMachineApiTest.java]: Add the other properties too?

**Comment by nacx** [comment]
Add also a live test for the two methods in the [VirtualMachineApiLiveTest](https://github.com/jclouds/jclouds/blob/master/apis/cloudstack/src/test/java/org/jclouds/cloudstack/features/VirtualMachineApiLiveTest.java) class?

File: apis/cloudstack/src/test/java/org/jclouds/cloudstack/features/VirtualMachineApiTest.java (hunk 1 above)
Comment (momania) [inline apis/cloudstack/src/test/java/org/jclouds/cloudstack/features/VirtualMachineApiTest.java]: Sure, can do that. Saw some other tests also tested just one property, but I'll adjust it.

**Comment by momania** [comment]
Ah, missed that one, will add that too.

**Comment by momania** [comment]
@nacx I've added a simple live test to test the virtual machine update. The test of service offering change is a bit more tricky. Currently only one template is picked and I don't know if on the cloudstack infra that runs behind the CI has more templates available. Checking for unknown properties would mean a lot of try-push-wait to the CI for me.

**Comment by nacx** [comment]
Squashed and pushed to [master](http://git-wip-us.apache.org/repos/asf/jclouds/commit/9be7339c) and [1.9.x](http://git-wip-us.apache.org/repos/asf/jclouds/commit/f69518b7). Thanks @momania!

**Pull request closed without merging**